### PR TITLE
IPCs can die from low power

### PR DESCRIPTION
**Pull request by BiancaWilkson**
## What Does This PR Do

IPCs now begin taking damage to their microbattery when their power is low enough. Most of the code was taken from #17993 
## Why It's Good For The Game

It adds consistency to the game, and means IPCs need to actually stay charged like how humans need to actually eat. It will kill you about as quickly as starvation does, so you get plenty of time to fix your power issues.

## Testing

I locked binary bot in a box and he died, then I made sure human nutrition didn't break too
## Changelog
:cl:
tweak: IPCs can die from starvation
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species/machine.dm [hunk 1]
```diff
@@ -113,6 +113,23 @@
 			H.update_hair()
 			H.update_fhair()
 
+/datum/species/machine/handle_life(mob/living/carbon/human/H) //KittyNoodle's code from PR #17993, handles IPC starvation
+	if(isLivingSSD(H)) // We don't want AFK people dying from this
+		return
+	if(H.nutrition < NUTRITION_LEVEL_HYPOGLYCEMIA)
+		if(prob(6))
+			H.adjustBrainLoss(4)
+			to_chat(H, "<span class='warning'> Error 74: Positronic brain receiving insufficient power.</span>")
```
Comment (Henri215) [inline code/modules/mob/living/carbon/human/species/machine.dm:122]:
```suggestion
			to_chat(H, "<span class='warning'>Error 74: Positronic brain receiving insufficient power.</span>")
```

**Comment by Mitchs98** [comment]
I'm really not sure if brain damage is the best way to handle this. Brute damage or stamina may be a better way to go about it to simulate the chassis locking up? Just makes no sense to me that the positronic works fine outside of the chassis yet would somehow need power while in the chassis.

Though I do agree consistency for needing to "eat" is good.

**Comment by BiancaWilkson** [comment]
> I'm really not sure if brain damage is the best way to handle this. Brute damage or stamina may be a better way to go about it to simulate the chassis locking up? Just makes no sense to me that the positronic works fine outside of the chassis yet would somehow need power while in the chassis.

I'm glad you made this comment because in hindsight it feels obvious to make the microbattery the organ that takes damage during a power failure.

**Comment by Mitchs98** [comment]
> > I'm really not sure if brain damage is the best way to handle this. Brute damage or stamina may be a better way to go about it to simulate the chassis locking up? Just makes no sense to me that the positronic works fine outside of the chassis yet would somehow need power while in the chassis.
> 
> I'm glad you made this comment because in hindsight it feels obvious to make the microbattery the organ that takes damage during a power failure.

Didn't think about that myself, but that does make much more sense with the chassis putting strain on the battery to try and forcibly power itself or what not!

File: code/modules/mob/living/carbon/human/species/machine.dm [hunk 2]
```diff
@@ -113,6 +113,23 @@
 			H.update_hair()
 			H.update_fhair()
 
+/datum/species/machine/handle_life(mob/living/carbon/human/H) //KittyNoodle's code from PR #17993, handles IPC starvation
+	if(isLivingSSD(H)) // We don't want AFK people dying from this
+		return
+	if(H.nutrition < NUTRITION_LEVEL_HYPOGLYCEMIA)
```
Comment (Contrabang) [inline code/modules/mob/living/carbon/human/species/machine.dm:119]:
```suggestion
	if(H.nutrition >= NUTRITION_LEVEL_HYPOGLYCEMIA)
		return
```

File: code/modules/mob/living/carbon/human/species/machine.dm [hunk 3]
```diff
@@ -113,6 +113,23 @@
...
+			to_chat(H, "<span class='warning'>Error 74: Positronic brain receiving insufficient power.</span>")
+		if(prob(4))
+			H.Weaken(6 SECONDS)
+			H.Stuttering(20 SECONDS)
+			to_chat(H, "<span class='warning'>Power critical, redirecting all power to positronic brain.</span>")
+			H.emote("collapse")
+			H.adjustBrainLoss(2)
+		if(prob(4))
+			to_chat(H, "<span class='warning'>Power critical, redirecting power from servos.</span>")
+			H.Slowed(rand(15 SECONDS, 32 SECONDS))
```
Comment (Contrabang) [inline code/modules/mob/living/carbon/human/species/machine.dm:131]: IMO, these should be else ifs. These are going to be firing off every 2 seconds, so theres a non-zero chance that multiple of these get hit at the same time
```suggestion
		if(prob(6))
			H.adjustBrainLoss(4)
			to_chat(H, "<span class='warning'>Error 74: Positronic brain receiving insufficient power.</span>")
		else if(prob(4))
			H.Weaken(6 SECONDS)
			H.Stuttering(20 SECONDS)
			to_chat(H, "<span class='warning'>Power critical, redirecting all power to positronic brain.</span>")
			H.emote("collapse")
			H.adjustBrainLoss(2)
		else if(prob(4))
			to_chat(H, "<span class='warning'>Power critical, redirecting power from servos.</span>")
			H.Slowed(rand(15 SECONDS, 32 SECONDS))
```

File: code/modules/mob/living/carbon/human/species/machine.dm [hunk 4]
```diff
@@ -113,22 +113,27 @@
 			H.update_hair()
 			H.update_fhair()
 
-/datum/species/machine/handle_life(mob/living/carbon/human/H) //KittyNoodle's code from PR #17993, handles IPC starvation
+/datum/species/machine/handle_life(mob/living/carbon/human/H) //Mostly KittyNoodle's code from PR #17993, handles IPC starvation
```
Comment (Contrabang) [inline code/modules/mob/living/carbon/human/species/machine.dm:116]:
```suggestion
/datum/species/machine/handle_life(mob/living/carbon/human/H) // Handles IPC starvation
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
We really shouldnt be mentioning the github really in the code

File: code/modules/mob/living/carbon/human/species/machine.dm [hunk 5]
```diff
@@ -113,6 +113,28 @@
...
+		microbattery.receive_damage(4, TRUE)
+	else if(prob(4))
+		H.Weaken(6 SECONDS)
+		H.Stuttering(20 SECONDS)
+		to_chat(H, "<span class='warning'>Power critical, shutting down superfluous functions.</span>")
+		H.emote("collapse")
+		microbattery.receive_damage(2, TRUE)
+	else if(prob(4))
+		to_chat(H, "<span class='warning'>Redirecting excess power from servos to vital components.</span>")
+		H.Slowed(rand(15 SECONDS, 32 SECONDS))
```
Comment (Contrabang) [inline code/modules/mob/living/carbon/human/species/machine.dm:136]: Arent you already slowed if you're starving, even as an IPC?

File: code/modules/mob/living/carbon/human/species/machine.dm [hunk 6]
```diff
@@ -113,6 +113,28 @@
 			H.update_hair()
 			H.update_fhair()
 
+/datum/species/machine/handle_life(mob/living/carbon/human/H) // Handles IPC starvation
+	if(isLivingSSD(H)) // We don't want AFK people dying from this
+		return
+	if(H.nutrition >= NUTRITION_LEVEL_HYPOGLYCEMIA)
+		return
+
+	var/obj/item/organ/internal/cell/microbattery = H.get_organ_slot("heart")
```
Comment (Contrabang) [inline code/modules/mob/living/carbon/human/species/machine.dm:122]: Also, can you reword your PR? The microbattery isn't the brain organ. This is technically heart damage

**Review by lewcc — APPROVED** [review]

**Pull request merged**